crowbar_framework/app/models/backup.rb

```diff
@@ -102,7 +102,14 @@ def upgrade?
   def upgrade
     upgrade = Crowbar::Upgrade.new(self)
     if upgrade.supported?
-      upgrade.upgrade
+      begin
+        upgrade.upgrade
+      rescue StandardError => e
```

Comment:
I would probably have put the `rescue` for the whole method, since the initializer for `Crowbar::Upgrade` already extracts the data.